content/error-handling/error-codes.textile

```diff
@@ -0,0 +1,301 @@
+
+
+h3(#90001). 90001 - Channel operation failed (invalid channel state)
+
+This error occurs when you attempt to operate on a channel that is in a state that does not permit the operation.
+
+An example of this error is when an application tries to "publish":/channels#publish a message to a channel that is in a failed state due to a prior error. As a result, the operation fails because actions cannot be performed on a channel in this state.
+
+*Resolution:* Ensure the channel is in an appropriate state before performing any operations. Use the Ably SDK to monitor and handle channel "state":/channels#listen-for-state changes effectively.
+
+h3(#90001). 90007 - Channel operation failed (invalid channel state)
+
+This error occurs when you attempt to operate on a channel that is in a "state":/channels#states that does not permit the operation.
+
+An example of this error is when an application tries to publish a message to a channel that is in a failed state due to a prior error. The operation fails because actions cannot be performed on a channel in this state.
+
+*Resolution:* Ensure the channel is in an appropriate state before performing any operations. Use the Ably SDK to monitor channel state changes and handle transitions effectively.
+
+h3(#90007). 90007 - Channel didn't attach within 10 seconds
+
+This error occurs when a channel fails to attach within 10 seconds. Clients with poor internet connections are more likely to encounter this issue because the ACK response to an ATTACH request does not arrive within the default timeout period.
+
+An example of this error is when a client reconnects after a disconnection and attempts to resume a channel's state. Messages sent during the period of disconnection have expired, and the system cannot guarantee continuity.
+
+*Resolution:* Adjust the @realtimeRequestTimeout@ or @channelRetryTimeout@ to a higher value. Note that this error was incorrectly assigned to error code 91200 in older versions of the Ably Pub/Sub Java SDK.
+
+h3(#91003). 91003 - Maximum member limit exceeded
+
+This error occurs when the maximum member limit for a channel has been exceeded, preventing additional clients from joining.
+
+An example of this error is An application attempting to add a new client to the "presence":/docs/presence-occupancy/presence?lang=javascript set of a channel, but the presence member limit for example, 100 members for the current subscription plan, has already been reached. The error prevents the additional client from joining.
+
+*Resolution:* Use "batching":/channels#batch-publish to optimize how presence messages are sent and received. This can help reduce the number of individual presence updates on the channel. You can also check your channel "limits":g/pricing/limits to understand the presence limits for your current plan.
```

Comment:
```suggestion
*Resolution:* Use "batching":/channels#batch-publish to optimize how presence messages are sent and received. This can help reduce the number of individual presence updates on the channel. You can also check your channel "limits":/pricing/limits to understand the presence limits for your current plan.
```

This should be recommending server-side batching to increase the limits, not batch presence.